File: sofaamy-bms/backend/seed.py
"""Populate the SQLite database with demo data.

Run once:  python seed.py
WARNING: drops and recreates all tables.
"""
from datetime import datetime, timedelta

from app.database import Base, engine, SessionLocal
from app import models
from app.lifecycle import ensure_engine_materials, STAGE_PROGRESS

Base.metadata.drop_all(bind=engine)
Base.metadata.create_all(bind=engine)
db = SessionLocal()
now = datetime.utcnow()
ago = lambda **kw: now - timedelta(**kw)

users = [
    ("Kwame Mensah", "supervisor", "+233240001122"),
    ("Yaa Boateng", "supervisor", "+233201112233"),
    ("Kofi Adjei", "rep", "+233272223344"),
    ("Yaw Darko", "qa", "+233554445566"),
    ("Esi Quaye", "accounts", "+233265556677"),
    ("Kojo Antwi", "procurement", "+233246667788"),
]
db.add_all([models.User(name=n, role=r, phone=p) for n, r, p in users])

clients = [
    ("Adom Estates Ltd", "Ama Owusu", "+233241182204", "East Legon, Accra", "company"),
    ("Golden Tulip Hotel", "Facilities", "+233302213344", "Airport City, Accra", "company"),
    ("Nii Armah Residence", "Nii Armah", "+233205559081", "Cantonments, Accra", "individual"),
    ("Trasacco Valley", "Project Team", "+233249007788", "Trasacco, Accra", "company"),
    ("Kumasi City Mall", "Ops Manager", "+233322084455", "Kumasi", "company"),
    ("Mensah Family Home", "Efua Mensah", "+233276641120", "Spintex, Accra", "individual"),
]
client_rows = [models.Client(name=n, contact=c, phone=p, location=l, type=t)
               for n, c, p, l, t in clients]
db.add_all(client_rows)

# legacy generic materials + every code the engines reference
materials = [
    ("AL-PRO-40", "Aluminium Profile 40mm", "Profile", "m", 85, 420, 200),
    ("AL-PRO-60", "Aluminium Profile 60mm", "Profile", "m", 110, 180, 200),
    ("GL-CLR-6", "Clear Glass 6mm", "Glass", "m2", 120, 64, 40),
    ("HW-HNG-01", "Casement Hinge", "Hardware", "pcs", 45, 12, 40),
    ("AC-GSK-01", "EPDM Rubber Gasket", "Accessory", "m", 6, 800, 300),
]
db.add_all([models.Material(code=c, name=n, category=cat, unit=u,
            unit_price=pr, stock=s, reorder_level=r)
            for c, n, cat, u, pr, s, r in materials])
db.commit()
ensure_engine_materials(db)

# jobs across the whole pipeline — value + staged payments tell the story
jobs = [
    # job_no, client_idx, product, stage, value, created days ago
    # (082's value = the frameless engine's real total for its attached design)
    ("SOF-2026-082", 0, "Frameless Swing Doors ×2", "pending",  12417.60, 0),
    ("SOF-2026-081", 0, "Sliding Windows ×12",      "assembly", 86400, 9),
    ("SOF-2026-080", 1, "Curtain Wall",             "cutting",  214000, 5),
    ("SOF-2026-079", 3, "Partition + Glass",        "glazing",  112800, 12),
    ("SOF-2026-078", 2, "Aluminium Door ×2",        "qa",       24600, 16),
    ("SOF-2026-077", 4, "Frameless Shopfront",      "dispatch", 96500, 21),
    ("SOF-2026-076", 5, "Casement Windows ×6",      "done",     31200, 30),
]
job_rows = {}
for jn, ci, pr, st, val, days in jobs:
    j = models.Job(job_number=jn, client_id=ci + 1, product=pr, stage=st,
                   progress=STAGE_PROGRESS.get(st, 0), value=val,
                   paid="0%", created_at=ago(days=days))
    job_rows[jn] = j
    db.add(j)
db.flush()

# attach the real configurator design to the pending job, so advancing it
# to Cutting issues glass + hardware from stock (the flagship demo path)
import json
swing_design = {
    "category": "frameless", "name": "Double Swing Doors", "ref": "AD-DR-1",
    "qty": 2, "location": "Main entrance, East Legon site",
    "system": "standard", "finishType": "powder",
    "width": 1900, "height": 2400, "cols": 2, "rows": 1, "frame": "mill",
    "colWidths": [950, 950], "rowHeights": [2400],
    "cells": [{"glass": "clear", "opening": "fixed", "panels": 1, "type": "door"},
              {"glass": "clear", "opening": "fixed", "panels": 1, "type": "door"}],
    "glassId": "temp10", "overPanel": False, "doorH": 2100,
    "flSystem": "klpatches", "slideSystem": "scl", "cornerAfter": -1,
    "scene": "shopfront",
}
db.add(models.DesignRecord(ref="AD-DR-1", name="Double Swing Doors",
                           client_name="Adom Estates Ltd", qty=2,
                           location="Main entrance, East Legon site",
                           total=12417.60, design_json=json.dumps(swing_design),
                           job_id=job_rows["SOF-2026-082"].id,
                           created_at=ago(days=1)))

# payments — deposits when jobs opened, balances near delivery (spread over
# the last 8 weeks so the revenue trend has shape)
payments = [
    ("SOF-2026-081", "deposit", "momo", 43200, 9),
    ("SOF-2026-080", "deposit", "bank", 107000, 5),
    ("SOF-2026-079", "deposit", "bank", 56400, 12),
    ("SOF-2026-078", "deposit", "momo", 12300, 16),
    ("SOF-2026-078", "balance", "momo", 12300, 2),
    ("SOF-2026-077", "deposit", "bank", 48250, 21),
    ("SOF-2026-077", "balance", "bank", 48250, 1),
    ("SOF-2026-076", "deposit", "cash", 15600, 30),
    ("SOF-2026-076", "balance", "momo", 15600, 24),
]
for jn, kind, method, amount, days in payments:
    db.add(models.Payment(job_id=job_rows[jn].id, kind=kind, method=method,
                          amount=amount, created_at=ago(days=days)))
for j in job_rows.values():
    paid = sum(a for jn2, _, _, a, _ in payments if jn2 == j.job_number)
    j.paid = f"{min(100, round(paid / j.value * 100)) if j.value else 0}%"

# QC + dispatch state
db.add(models.QcCheck(job_id=job_rows["SOF-2026-077"].id, result="pass", score=92,
                      inspector="Yaw Darko", notes="Edgework clean, dims within 1mm",
                      created_at=ago(days=2)))
db.add(models.QcCheck(job_id=job_rows["SOF-2026-079"].id, result="rework", score=71,
                      inspector="Yaw Darko", notes="Glazing bead gap on F2 — redo seal",
                      created_at=ago(days=1)))
job_rows["SOF-2026-077"].driver = "Sammy K."
job_rows["SOF-2026-077"].vehicle = "GT-4821-22"
job_rows["SOF-2026-077"].dn_number = "SOF-DN-2026-088"
job_rows["SOF-2026-076"].driver = "Emmanuel O."
job_rows["SOF-2026-076"].vehicle = "GS-1190-21"
job_rows["SOF-2026-076"].dn_number = "SOF-DN-2026-087"
job_rows["SOF-2026-076"].delivered_at = ago(days=23)

quotes = [
    ("SOF-Q-2026-0142", "Adom Estates Ltd", "Sliding Windows ×12", 86400, "Accepted", "SOF-2026-081", 10),
    ("SOF-Q-2026-0141", "Golden Tulip Hotel", "Curtain Wall", 214000, "Accepted", "SOF-2026-080", 6),
    ("SOF-Q-2026-0140", "Nii Armah Residence", "Aluminium Door ×2", 24600, "Accepted", "SOF-2026-078", 17),
    ("SOF-Q-2026-0139", "Trasacco Valley", "Partition + Glass", 112800, "Accepted", "SOF-2026-079", 13),
    ("SOF-Q-2026-0138", "Mensah Family Home", "Casement Windows ×6", 31200, "Declined", None, 8),
    ("SOF-Q-2026-0137", "Kumasi City Mall", "Frameless Sliding Partition", 42800, "Sent", None, 2),
    ("SOF-Q-2026-0136", "Adom Estates Ltd", "Frameless Swing Doors ×2", 12417.60, "Accepted", "SOF-2026-082", 1),
    ("SOF-Q-2026-0135", "Golden Tulip Hotel", "Balustrade Run — Pool Deck", 18900, "Draft", None, 0),
]
for qn, cn, pr, total, status, jn, days in quotes:
    db.add(models.Quote(quote_number=qn, client_name=cn, product=pr, total=total,
                        status=status, created_at=ago(days=days),
                        job_id=job_rows[jn].id if jn else None))

# activity feed
events = [
    (None, "Esi Quaye", "payment", "recorded balance of GHS 48,250.00 (bank) — SOF-2026-077", 1, "SOF-2026-077"),
    (None, "Yaw Darko", "qc", "SOF-2026-077 passed QA (92%) — Edgework clean, dims within 1mm", 2, "SOF-2026-077"),
    (None, "Kwame Mensah", "dispatch", "delivery SOF-DN-2026-088 assigned to Sammy K. (GT-4821-22)", 1, "SOF-2026-077"),
    (None, "Yaw Darko", "qc", "SOF-2026-079 flagged for REWORK at QA (71%) — Glazing bead gap on F2", 1, "SOF-2026-079"),
    (None, "Kwame Mensah", "quote", "sent quote SOF-Q-2026-0137 to Kumasi City Mall via WhatsApp", 2, None),
    (None, "Kwame Mensah", "quote", "quote SOF-Q-2026-0136 accepted — job SOF-2026-082 opened for Adom Estates Ltd (GHS 12,418), awaiting 50% deposit", 1, "SOF-2026-082"),
    (None, "Yaa Boateng", "stage", "moved SOF-2026-080 to Cutting — materials issued: 118.4 m AL-MOLLIUM, 86.2 m AL-TRANSUM", 4, "SOF-2026-080"),
    (None, "Kojo Antwi", "stock", "received 200 m AL-MOLLIUM (Mollium Profile)", 3, None),
]
for _, who, kind, note, days, jn in events:
    db.add(models.Event(job_id=job_rows[jn].id if jn else None, who=who, kind=kind,
                        note=note, created_at=ago(days=days, hours=2)))

# a few stock movements so Inventory has history
mat = {m.code: m for m in db.query(models.Material).all()}
moves = [
    ("AL-MOLLIUM", 200, "Goods received — Tema supplier", "", 3),
    ("AL-MOLLIUM", -118.4, "Issued to SOF-2026-080", "SOF-2026-080", 4),
    ("AL-TRANSUM", -86.2, "Issued to SOF-2026-080", "SOF-2026-080", 4),
    ("GL-TMP-10", -11.4, "Issued to SOF-2026-077", "SOF-2026-077", 18),
    ("BL 203", -22, "Issued to SOF-2026-077", "SOF-2026-077", 18),
]
for code, delta, reason, jn, days in moves:
    if code in mat:
        db.add(models.StockMove(material_id=mat[code].id, delta=delta, reason=reason,
                                job_number=jn, created_at=ago(days=days)))

db.commit()
print("Seeded sofaamy.db:",
      db.query(models.Client).count(), "clients,",
      db.query(models.Job).count(), "jobs,",
      db.query(models.Material).count(), "materials,",
      db.query(models.Quote).count(), "quotes,",
      db.query(models.Payment).count(), "payments,",
      db.query(models.Event).count(), "events.")
db.close()

File: frontend/src/data/seed.js
// ============================================================
// DEMO SEED DATA — realistic Ghana / GHS content so every screen
// looks live. Replaced by the SQLite backend once wired.
// ============================================================

export const COMPANY = {
  name: 'Sofaamy Co. Ltd',
  tagline: 'Glass & Aluminium Fabrication',
  location: 'Accra, Ghana',
  poweredBy: 'Veloxa Technology',
}

export const CURRENT_USER = { name:'Kwame Mensah', role:'Supervisor', initials:'KM' }

const av = ['#2471A3','#1E8449','#CA6F1E','#6C3483','#C0392B','#16A085','#8E44AD','#2E86C1']
export const avatarColor = (i) => av[i % av.length]

export const STAGES = [
  { key:'cutting',    label:'Cutting' },
  { key:'processing', label:'Processing' },
  { key:'holes',      label:'Holes / Routing' },
  { key:'assembly',   label:'Assembly' },
  { key:'glazing',    label:'Glazing' },
  { key:'qa',         label:'Quality Check' },
  { key:'dispatch',   label:'Dispatch' },
  { key:'install',    label:'Installation' },
]

export const KPIS = [
  { key:'revenue', label:'Revenue (This Month)', value:'₵486,200', trend:'+18.4%', dir:'up',   tone:'green'  },
  { key:'jobs',    label:'Active Jobs',          value:'34',       trend:'+6',     dir:'up',   tone:'blue'   },
  { key:'quotes',  label:'Open Quotations',      value:'19',       trend:'8 pending', dir:'flat', tone:'orange' },
  { key:'convert', label:'Quote → Order Rate',   value:'62%',      trend:'+4.1%',  dir:'up',   tone:'purple' },
]

// Revenue trend (last 8 weeks, GHS thousands)
export const REVENUE_TREND = [
  { label:'W1', value:62 }, { label:'W2', value:74 }, { label:'W3', value:58 },
  { label:'W4', value:91 }, { label:'W5', value:88 }, { label:'W6', value:104 },
  { label:'W7', value:97 }, { label:'W8', value:121 },
]

export const PIPELINE_MIX = [
  { label:'Cutting',    value:6,  color:'#2471A3' },
  { label:'Assembly',   value:9,  color:'#1E8449' },
  { label:'Glazing',    value:5,  color:'#CA6F1E' },
  { label:'Quality',    value:4,  color:'#6C3483' },
  { label:'Dispatch',   value:7,  color:'#16A085' },
  { label:'Install',    value:3,  color:'#C0392B' },
]

export const PRODUCT_DEMAND = [
  { label:'Sliding Windows', value:38 },
  { label:'Curtain Walls',   value:22 },
  { label:'Doors',           value:18 },
  { label:'Partitions',      value:12 },
  { label:'Balustrades',     value:10 },
]

export const CLIENTS = [
  { id:1, name:'Adom Estates Ltd',        contact:'Ama Owusu',    phone:'+233 24 118 2204', location:'East Legon, Accra',    jobs:5, value:'₵214,500', type:'company' },
  { id:2, name:'Nii Armah Residence',     contact:'Nii Armah',    phone:'+233 20 555 9081', location:'Cantonments, Accra',   jobs:2, value:'₵58,900',  type:'individual' },
  { id:3, name:'Golden Tulip Hotel',      contact:'Facilities',   phone:'+233 30 221 3344', location:'Airport City, Accra',  jobs:3, value:'₵402,000', type:'company' },
  { id:4, name:'Mensah Family Home',      contact:'Efua Mensah',  phone:'+233 27 664 1120', location:'Spintex, Accra',       jobs:1, value:'₵31,200',  type:'individual' },
  { id:5, name:'Trasacco Valley',         contact:'Project Team', phone:'+233 24 900 7788', location:'Trasacco, Accra',      jobs:4, value:'₵318,700', type:'company' },
  { id:6, name:'Kumasi City Mall',        contact:'Ops Manager',  phone:'+233 32 208 4455', location:'Kumasi',               jobs:2, value:'₵176,300', type:'company' },
]

export const LEADS = [
  { id:1, name:'Villa Project — Aburi',   contact:'Kofi Asante',  value:'₵120,000', stage:'Qualified',  source:'WhatsApp', owner:'Yaa Boateng',  age:'2d' },
  { id:2, name:'Office Fit-out — Ridge',  contact:'Sarah Tetteh', value:'₵260,000', stage:'Proposal',   source:'Referral', owner:'Kwame Mensah', age:'5d' },
  { id:3, name:'Apartment Block — Tema',  contact:'Ibrahim M.',   value:'₵540,000', stage:'Survey',     source:'Field Rep',owner:'Yaa Boateng',  age:'1d' },
  { id:4, name:'Retail Shopfront — Osu',  contact:'Grace A.',     value:'₵88,000',  stage:'New',        source:'Walk-in',  owner:'Kwame Mensah', age:'4h' },
  { id:5, name:'Hotel Balustrades',       contact:'Golden Tulip', value:'₵150,000', stage:'Negotiation',source:'WhatsApp', owner:'Kwame Mensah', age:'8d' },
]

export const QUOTES = [
  { id:'SOF-Q-2026-0142', client:'Adom Estates Ltd',  product:'Sliding Windows ×12', total:'₵86,400',  status:'Approved', date:'05 Jul 2026' },
  { id:'SOF-Q-2026-0141', client:'Golden Tulip Hotel',product:'Curtain Wall',        total:'₵214,000', status:'Sent',     date:'04 Jul 2026' },
  { id:'SOF-Q-2026-0140', client:'Nii Armah Residence',product:'Aluminium Door ×2',  total:'₵24,600',  status:'Draft',    date:'04 Jul 2026' },
  { id:'SOF-Q-2026-0139', client:'Trasacco Valley',   product:'Partition + Glass',   total:'₵112,800', status:'Approved', date:'02 Jul 2026' },
  { id:'SOF-Q-2026-0138', client:'Mensah Family Home',product:'Casement Windows ×6', total:'₵31,200',  status:'Rejected', date:'01 Jul 2026' },
  { id:'SOF-Q-2026-0137', client:'Kumasi City Mall',  product:'Frameless Shopfront', total:'₵96,500',  status:'Sent',     date:'30 Jun 2026' },
]

export const JOBS = [
  { id:'SOF-2026-081', client:'Adom Estates Ltd',   product:'Sliding Windows ×12', stage:'assembly', progress:55, due:'12 Jul', supervisor:'Kwame Mensah', paid:'50%' },
  { id:'SOF-2026-080', client:'Golden Tulip Hotel', product:'Curtain Wall',        stage:'cutting',  progress:15, due:'28 Jul', supervisor:'Yaa Boateng',  paid:'50%' },
  { id:'SOF-2026-079', client:'Trasacco Valley',    product:'Partition + Glass',   stage:'glazing',  progress:72, due:'10 Jul', supervisor:'Kwame Mensah', paid:'50%' },
  { id:'SOF-2026-078', client:'Nii Armah Residence',product:'Aluminium Door ×2',   stage:'qa',       progress:88, due:'08 Jul', supervisor:'Yaa Boateng',  paid:'100%' },
  { id:'SOF-2026-077', client:'Kumasi City Mall',   product:'Frameless Shopfront', stage:'dispatch', progress:94, due:'07 Jul', supervisor:'Kwame Mensah', paid:'100%' },
  { id:'SOF-2026-076', client:'Mensah Family Home', product:'Casement Windows ×6', stage:'install',  progress:98, due:'06 Jul', supervisor:'Yaa Boateng',  paid:'100%' },
]

export const SURVEYS = [
  { id:'SV-311', job:'SOF-2026-082', site:'Apartment Block — Tema', rep:'Kofi Adjei',  status:'Completed', when:'Today 09:20', units:14, variance:'+2.1%' },
  { id:'SV-310', job:'SOF-2026-080', site:'Airport City, Accra',   rep:'Kofi Adjei',  status:'Completed', when:'Yesterday',  units:8,  variance:'-0.4%' },
  { id:'SV-309', job:'—',           site:'Villa Project — Aburi',  rep:'Abena Sarpong',status:'Scheduled', when:'08 Jul 11:00',units:0,  variance:'—' },
  { id:'SV-308', job:'SOF-2026-079', site:'Trasacco, Accra',       rep:'Abena Sarpong',status:'Review',    when:'03 Jul',     units:6,  variance:'+5.0%' },
]

export const INVENTORY = [
  { code:'AL-PRO-40',  name:'Aluminium Profile 40mm', cat:'Profile',  stock:420, unit:'m',   reorder:200, price:'₵85/m',    status:'ok' },
  { code:'AL-PRO-60',  name:'Aluminium Profile 60mm', cat:'Profile',  stock:180, unit:'m',   reorder:200, price:'₵110/m',   status:'low' },
  { code:'GL-CLR-6',   name:'Clear Glass 6mm',        cat:'Glass',    stock:64,  unit:'m²',  reorder:40,  price:'₵120/m²',  status:'ok' },
  { code:'GL-TMP-8',   name:'Tempered Glass 8mm',     cat:'Glass',    stock:22,  unit:'m²',  reorder:30,  price:'₵230/m²',  status:'low' },
  { code:'HW-SLD-01',  name:'Sliding Roller Set',     cat:'Hardware', stock:96,  unit:'set', reorder:50,  price:'₵240/set', status:'ok' },
  { code:'HW-HNG-01',  name:'Casement Hinge',         cat:'Hardware', stock:12,  unit:'pcs', reorder:40,  price:'₵45/pc',   status:'critical' },
  { code:'AC-GSK-01',  name:'EPDM Rubber Gasket',     cat:'Accessory',stock:800, unit:'m',   reorder:300, price:'₵6/m',     status:'ok' },
]

export const DISPATCH = [
  { id:'DSP-091', job:'SOF-2026-077', client:'Kumasi City Mall',  units:6, driver:'Sammy K.',  vehicle:'GT-4821-22', status:'In Transit', qr:'scanned' },
  { id:'DSP-090', job:'SOF-2026-078', client:'Nii Armah Residence',units:2,driver:'Emmanuel O.',vehicle:'GS-1190-21', status:'Delivered',  qr:'scanned' },
  { id:'DSP-089', job:'SOF-2026-076', client:'Mensah Family Home',units:6, driver:'Sammy K.',  vehicle:'GT-4821-22', status:'Installed',  qr:'scanned' },
  { id:'DSP-088', job:'SOF-2026-079', client:'Trasacco Valley',   units:4, driver:'—',          vehicle:'—',          status:'Scheduled',  qr:'pending' },
]

export const QA_CHECKS = [
  { id:'QC-204', job:'SOF-2026-078', product:'Aluminium Door ×2',  stage:'Post-Production',  inspector:'Yaw Darko', result:'Pass',   score:'96%', when:'Today' },
  { id:'QC-203', job:'SOF-2026-077', product:'Frameless Shopfront',stage:'Post-Production',  inspector:'Yaw Darko', result:'Pass',   score:'92%', when:'Yesterday' },
  { id:'QC-202', job:'SOF-2026-079', product:'Partition + Glass',  stage:'Glazing Check',    inspector:'Yaw Darko', result:'Rework', score:'71%', when:'2 Jul' },
  { id:'QC-201', job:'SOF-2026-076', product:'Casement Windows ×6',stage:'Post-Installation',inspector:'Yaw Darko', result:'Pass',   score:'98%', when:'1 Jul' },
]

export const TEAM = [
  { name:'Kwame Mensah', role:'Supervisor',  jobs:12, phone:'+233 24 000 1122' },
  { name:'Yaa Boateng',  role:'Supervisor',  jobs:9,  phone:'+233 20 111 2233' },
  { name:'Kofi Adjei',   role:'Field Rep',   jobs:'—',phone:'+233 27 222 3344' },
  { name:'Abena Sarpong',role:'Field Rep',   jobs:'—',phone:'+233 24 333 4455' },
  { name:'Yaw Darko',    role:'Quality (QA)',jobs:'—',phone:'+233 55 444 5566' },
  { name:'Esi Quaye',    role:'Accounts',    jobs:'—',phone:'+233 26 555 6677' },
  { name:'Kojo Antwi',   role:'Procurement', jobs:'—',phone:'+233 24 666 7788' },
]

export const ACTIVITY = [
  { who:'Yaa Boateng',  what:'moved job SOF-2026-080 to Cutting',        when:'12m ago',  tone:'blue' },
  { who:'System',       what:'sent quote SOF-Q-2026-0142 via WhatsApp',  when:'40m ago',  tone:'green' },
  { who:'Kofi Adjei',   what:'submitted survey SV-311 (14 units)',       when:'1h ago',   tone:'purple' },
  { who:'Esi Quaye',    what:'confirmed 50% deposit — Adom Estates',     when:'2h ago',   tone:'gold' },
  { who:'Yaw Darko',    what:'flagged SOF-2026-079 for rework at Glazing',when:'3h ago',  tone:'red' },
  { who:'Kwame Mensah', what:'approved quote for Trasacco Valley',       when:'5h ago',   tone:'blue' },
]

export const statusTone = (s) => ({
  Approved:'b-green', Sent:'b-blue', Draft:'b-gray', Rejected:'b-red',
  Pass:'b-green', Rework:'b-orange', Fail:'b-red',
  Completed:'b-green', Scheduled:'b-blue', Review:'b-orange',
  Delivered:'b-green', Installed:'b-green', 'In Transit':'b-blue',
  ok:'b-green', low:'b-orange', critical:'b-red',
  New:'b-gray', Qualified:'b-blue', Survey:'b-purple', Proposal:'b-orange', Negotiation:'b-gold',
}[s] || 'b-gray')
